lib/ansible/modules/cloud/amazon/aws_guardduty_detector.py

```diff
@@ -0,0 +1,157 @@
+  guardduty_detector:
+    state: absent
+'''
+
+
+RETURN = r'''
+detector_id:
+    description: The ID of the GuardDuty detector you just created or updated.
+    returned: always
+    type: string
+'''
+
+import os
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+
+try:
+    from botocore.exceptions import BotoCoreError, ClientError
+except ImportError:
+    pass  # handled by AnsibleAWSModule
+
+
+def detector_exists(client, module, result):
+    try:
+        response = client.list_detectors()
+        return {'exists': True, 'detector_id': response['DetectorIds'][0]}
+    except (ClientError, IndexError):
+        return {'exists': False}
+
+    return {'exists': False}
+
+
+def create_detector(client, module):
+    if module.check_mode:
+        module.exit_json(changed=True)
+    try:
+        response = client.create_detector(
+            Enable=module.params.get('enabled')
+        )
+        return {'changed': True, 'detector_id': response['DetectorId']}
+    except (BotoCoreError, ClientError) as e:
+        module.fail_json_aws(e, msg="Failed to enable detector")
+
+
+def update_detector(client, module, status):
+    if module.check_mode:
+        module.exit_json(changed=True)
+    try:
+        response = client.update_detector(
+            DetectorId=status['detector_id'],
+            Enable=module.params.get('enabled')
+        )
+        return {'changed': True, 'detector_id': status['detector_id']}
+    except (BotoCoreError, ClientError) as e:
+        module.fail_json_aws(e, msg="Failed to update detector")
+
+
+def delete_detector(client, module, status):
+    if module.check_mode:
+        module.exit_json(changed=True)
+    try:
+        response = client.delete_detector(
+            DetectorId=status['detector_id']
+        )
+        return {'changed': True, 'detector_id': status['detector_id']}
+    except (BotoCoreError, ClientError) as e:
+        module.fail_json_aws(e, msg="Failed to delete detector")
+
+
+def main():
+    module = AnsibleAWSModule(
+        argument_spec={
+            'state': dict(type='str', choices=['present', 'absent'], required=True),
+            'enabled': dict(type='bool', default=True),
+        },
+        supports_check_mode=True,
+    )
+
+    result = {
+        'changed': False,
+        'detector_id': ''
+    }
+
+    desired_state = module.params.get('state')
+
+    client = module.client('guardduty')
+
+    detector_status = detector_exists(client, module, result)
+
+    if desired_state == 'present':
+        if not detector_status['exists']:
+            result = create_detector(client, module)
+        if detector_status['exists']:
+            result = update_detector(client, module, detector_status)
+
+    if desired_state == 'absent':
+        if detector_status['exists']:
+            result = delete_detector(client, module, detector_status)
+
+    module.exit_json(changed=result['changed'], detector_id=result['detector_id'])
```

Comment:
It would be nice to add for `present` state to the returns the status from [get_detector](http://boto3.readthedocs.io/en/latest/reference/services/guardduty.html#GuardDuty.Client.get_detector) for the service role ARN and the status that AWS has for the detector.